src/document.rs

```diff
@@ -328,7 +332,7 @@ impl Document {
             let excerpt = self.render_html(excerpt_str, globals, parser, markdown)?;
             Value::scalar(excerpt)
         } else if self.front.excerpt_separator.is_empty() {
-            Value::Nil
+            return Ok(());
```

Comment:
When writing code, I tend to prefer people making conditionals off of the value of a variable rather than the variable existing or not (like bash or make).  I think my intention of making this `Nil` was so that a `{{ page.excerpt }}` variable would always exist but it would just be `nil`.

The other downside to this route is it is a breaking change.

An alternative is for us to instead change the code that is reading this variable and check `value.is_nil()`.

Reply:
> When writing code, I tend to prefer people making conditionals off of the value of a variable rather than the variable existing or not (like bash or make).
I see, I didn't even think of that being exposed as a variable to the user, to be honest -- I just considered the Rust "layer" of things.

> I think my intention of making this `Nil` was so that a `{{ page.excerpt }}` variable would always exist but it would just be `nil`.
> 
> The other downside to this route is it is a breaking change.
> 
> An alternative is for us to instead change the code that is reading this variable and check `value.is_nil()`.

Sounds like the reasonable solution, then.

Reply:
> I see, I didn't even think of that being exposed as a variable to the user, to be honest -- I just considered the Rust "layer" of things.

I'm a little rough on this code since looking at a bunch of other things since then but I believe this is populating the variables available for a page in liquid.  We just happen to reuse it within Rust.  The hint towards this is that it is using a `liquid::Value` rather than keeping things more rust-native (Option<String>).